pkg/apiproxy/ytsaurus.go

```diff
@@ -115,6 +115,17 @@ func (c *Ytsaurus) SaveClusterState(ctx context.Context, clusterState ytv1.Clust
 	return nil
 }
 
+func (c *Ytsaurus) SaveObserverGeneration(ctx context.Context, generation int64) error {
+	logger := log.FromContext(ctx)
+	c.ytsaurus.Status.ObservedGeneration = generation
+	if err := c.apiProxy.UpdateStatus(ctx); err != nil {
+		logger.Error(err, "unable to update Ytsaurus cluster status")
```

Comment:
I think we should either log error or return, otherwise we have logged it twice.
If you want to add extra context, I suggest to wrap the error with `return fmt.Errorf("failed to set status: %w", err)`